### PR TITLE
Clarify box.runtime.info() description

### LINKED ISSUE #2911: feedback: box.runtime.info() | Tarantool
> <…>lloc is the maximal memory quota that can be allocated for Lua;
> **|used is the current memory size used by Lua|.**
> 
> Return type:table
> 
> Example:
> box.runtime.info()
> box.runtime.in<…>

https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_slab/runtime_info/

As I (@Totktonada) see from the code, it has no relation to Lua memory.

---
@patiencedaur: clarify the difference between "Lua memory" and "memory occupied by certain Tarantool Lua objects"

### PR DESCRIPTION
**Pull request by patiencedaur**
Resolves #2911

### CODE REVIEW

File: doc/reference/reference_lua/box_slab/runtime_info.rst [hunk 1]
```diff
@@ -8,13 +8,15 @@ box.runtime.info()
 
 .. function:: box.runtime.info()
 
-    Show a memory usage report (in bytes) for the Lua runtime.
+    Show a memory usage report (in bytes) for the runtime arena.
+    The runtime arena encompasses internal Lua memory as well as network buffers
+    and other memory expenses associated with the application server subsystem.
```
Comment (Totktonada) [inline doc/reference/reference_lua/box_slab/runtime_info.rst]: I found 'Lua memory' confusing in this context. One can understand it as 'memory, where all those table/string/number content is stored'. I can propose an attempt to explain it, but, please, fact-check it with Alexander L.

> memtx stores tuples, indexes, transactions in its own memory arena. The same about vinyl: it stores various caches in its own arena. However there is the special arena called 'runtime arena' for data that is not related neither to memtx, nor to vinyl. Here tarantool stores data associated with the application server subsystem. For example, buffers referenced from Lua and tuples created by `box.tuple.new()`.

**Comment by Totktonada** [comment]
@DifferentialOrange Please, look, whether your questions from the [discussion](https://github.com/orgs/tarantool/discussions/7106) were resolved here.

File: doc/reference/reference_lua/box_slab/runtime_info.rst (hunk 1 above)
Comment (patiencedaur) [reply doc/reference/reference_lua/box_slab/runtime_info.rst]: Where is this quote from?

Comment (Totktonada) [reply doc/reference/reference_lua/box_slab/runtime_info.rst]: Ouch, sorry, I forgot to clarify it. It is how everything works in my head.

**Review by alyapunov — APPROVED** [review]

File: doc/reference/reference_lua/box_slab/runtime_info.rst (hunk 1 above)
Comment (Totktonada) [reply doc/reference/reference_lua/box_slab/runtime_info.rst]: I still think that 'runtime memory encompasses Lua memory' is confusing.

Comment (patiencedaur) [reply doc/reference/reference_lua/box_slab/runtime_info.rst]: It's how Alexander L. explained it. I like your explanation, but I doubt that this reference page is the best place for it.

**Comment by DifferentialOrange** [comment]
> @DifferentialOrange Please, look, whether your questions from the [discussion](https://github.com/orgs/tarantool/discussions/7106) were resolved here.

Yeah, thank you!

**Pull request merged**